mplc/multi_partner_learning.py

```diff
@@ -603,12 +605,65 @@ def fit_minibatch(self):
         logger.debug("End of S-Model collaborative round.")
 
 
-# Supported multi-partner learning approaches
+class EnsemblePredictions(MultiPartnerLearning):
+    """
+    Ensemble (average) prediction of several input models
+    This approach can only be used with the EnsemblePredictionsModel
+    """
+
+    def __init__(self, scenario, **kwargs):
+        # First, if only one partner, fall back to dedicated single partner function
+        super(EnsemblePredictions, self).__init__(scenario, **kwargs)
+
+        if self.partners_count == 1:
+            raise ValueError('Only one partner is provided. Please use the dedicated SinglePartnerLearning class')
+
+        partner_model_list = [self.dataset.generate_new_model() for _ in range(self.partners_count)]
+        self.model = EnsemblePredictionsModel(partner_model_list)
+
+        for partner in self.partners_list:
+            partner.model_weights = self.model_weights
+        # the partners should all start with a different set of weights
+        logger.info("Init EnsemblePredictionsModel model")
+
+    def build_model(self):
+        partner_model_list = [partner.build_model() for partner in self.partners_list]
+        return EnsemblePredictionsModel(partner_model_list)
+
+    def fit_epoch(self):
+        # Clear Keras' old models
+        clear_session()
 
+        self.eval_and_log_model_val_perf()
+
+        for partner_index, partner in enumerate(self.partners_list):
+
+            partner_model = partner.build_model()  # issue here is it tries to build
+            # the partner model with the Ensembling model
```

Comment:
I don't understand one thing: at the beginning of an epoch here, we are supposed to init the local models with the resulting joint model from the previous epoch, right? How is it achieved here?

Reply:
No, we don't init the local models with the resulting joint model from the previous epoch. Every model training is done independantly. 
You only average the predictions at the very end of the training.

Reply:
Oh OK 🤦‍♂️ ! Then follow-up questions: how come you don't pass the `fit()` any parameter for the nb of epochs to train on? Do you rebuild a model for each epoch - which wouldn't make much sense for local trainings right?

Reply:
Yes but this is inside the fit_epoch() function so it should be for one epoch only.
Another solution would be to rewrite the fit() method of MultipartnerLearning for this subclass.